### ISSUE TITLE
Typing space fast prints out a dot?

### ISSUE THREAD

**Issue by octref** [VSCodeVim/Vim#2108]
* Click *thumbs-up* 👍 on this issue if you want it!
* Click *confused* 😕 on this issue if not having it makes VSCodeVim unusable. 

The VSCodeVim team prioritizes issues based on reaction count. 

--------

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): Bug

**Environment**:

- **VSCode Version**: 1.17
- **VsCodeVim Version**: 0.10.2
- **OS**: macOS 10.12.6

**What happened**:

Type something like `code` or any other words, then type a few spaces promptly.

![type](https://user-images.githubusercontent.com/4033249/32299773-0c08c500-bf14-11e7-88c7-f8fb2699e792.gif)

**Comment by Chillee** [comment]
I can't reproduce it. Is it possible it's some other extension interacting weirdly?

**Comment by octref** [comment]
When I disable Vim this no longer repros. Let me try again...

**Comment by octref** [comment]
I disabled all other extensions, still happens.

![type](https://user-images.githubusercontent.com/4033249/32300156-75a2dee6-bf15-11e7-8220-1b51c28cf923.gif)

**Comment by Chillee** [comment]
lol this is bizarre. Do you have anything in your keybindings.json or settings.json that might cause something like this? 

Also, is this something that broke recently, or are you installing VSCodeVim just now?

**Comment by octref** [comment]
I have been using it very long, this started to happen probably since a month ago.

Just tried moving `~/.vscode` to `~/.vscode-backup`, and installed Vim. Seems not happening anymore, so there must be something else I guess..

**Comment by octref** [comment]
It's still a mystery to me, but it's happening in other programs for me too so I guess it's not related to VSCodeVim.

**Comment by octref** [comment]
OK, it's a mac thing...

![image](https://user-images.githubusercontent.com/4033249/33396153-8b002e5e-d4fc-11e7-9c38-fb4f71fcc303.png)

**Comment by Chillee** [comment]
lol wtf

**Comment by octref** [comment]
Yeah it was on. Have no idea why they would include this on macOS and turn it on by default...

**Comment by shoeffner** [comment]
The settings panel changed in recent macOS versions: https://apple.stackexchange.com/a/451369 – it is now available under "System Settings -> Keyboard -> Text Input -> Input Sources -> Edit... -> Add full stop with double-space".

**Comment by lwerdna** [comment]
@shoeffner Thank you 👍

**Comment by dongli** [comment]
This Apple thing drove me crazy! Thanks for pointing the reason out!
